### ISSUE TITLE
OSM Toilets exist

### ISSUE THREAD

**Issue by Cj-Malone** [sozialhelden/wheelmap-frontend#261]
Sorry if this is the wrong repo.

A growing number of OSM POIs are now tagged with `toilets:wheelchair=no`, but are missing `toilets=yes` or `amenity=toilets`. It's impossible for an client of OSM to know whether there are toilets, just not accessible, or there aren't toilets at all. The inverse is also true, you get `toilets:wheelchair=yes` without `toilets=yes`/`amenity=toilets` so the client might not show that a POI would have any toilets, accessible or not.

This of cause leads to some odd issues, like a [bus stop](https://www.openstreetmap.org/node/471490746), without disabled toilets, but might have actual toilets. We of cause know that it should have `toilets=no` too, but a client may not. And here is a [church](https://www.openstreetmap.org/node/1975010764), we don't know if it has toilets, but it does have disabled toilets.

There are thousands of occurrences of this, you can check with overpass:
```
nwr["toilets:wheelchair"="no"][!"toilets"]
```

It would be great if when collecting data from users you could ask them, and then add the tag to OSM.

**Comment by opyh** [comment]
Thanks for your input!

The first case you've linked is a bit older, this wouldn't happen with the current implementation anymore.

Current logic:

- We don't allow certain place types, like bus stops, to be tagged with `toilets:wheechair` anymore. Wheelmap displays a link to the next accessible toilet anyway since a while ago, even if it belongs to a different PoI - so you can still find it when starting from a bus stop.
- We require a place to be tagged as wheelchair accessible before you can tag if it has an accessible toilet. This seems like a sensible default as there are real-world cases of 'accessible' toilets hidden behind non-accessible stairs and entrances, but showing them in the UI causes more confusion than it helps anyone.

We're currently working on an overhaul of the whole OSM backend to allow more complex attribute tagging, and to sync the more complex 'Add a place' survey content with other sensible OSM tags. This will allow us to add `toilets=yes` when a user adds a toilet in a survey. The whole project involves a huge effort with a lot of refactoring, so I hope you understand the current behavior will have to stay as it is for a while. I'll leave the issue open until have a better means to fix it.

**Comment by Cj-Malone** [comment]
> the current behavior will have to stay as it is for a while

Yeah, that's fine, I just wanted to make sure you were aware of the problem.

I think your current workflow is good, but adding 1 question after the user says the toilets are not accessible might be the way to go:
```
Wheelchair accessible?
|-No
 |-Send OSM `wheelchair=no`
|-Yes/Limited
 |-Accessible toilets?
  |-Yes
    |-Send to OSM `wheelchair=yes toilets=yes toilets:wheelchair=yes`
  |-No
+  |-Are there inaccessible toilets?
+   |-No
+    |-Send to OSM `wheelchair=yes toilets=no`
+   |-Yes
+    |-Send to OSM `wheelchair=yes toilets=yes toilets:wheelchair=no`
```

Or you could tri state the "Accessible toilets?" question into:
Yes, toilets are accessible. (`wheelchair=yes toilets=yes toilets:wheelchair=yes`)
No, there are toilets but they are inaccessible. (`wheelchair=yes toilets=yes toilets:wheelchair=no`)
No, there are no toilets at all. (`wheelchair=yes toilets=no`)

Anyway good look with the refactoring, this is a great project. :)

**Comment by opyh** [comment]
Thanks for the detailled suggestion! I'll consider your flow when coming back to this feature's code again.
